### DISCUSSION TITLE
Warehouse

### DISCUSSION THREAD

**Discussion by D4RKNOISE** [Budibase/budibase#2601] [Help]
Hey,

https://i.imgur.com/gHg94dY.png

I'm trying to create a user dashboard where after the user logs in, only his data can be seen.

Goal: writing field logs and managing his material.

I'm a bit stuck on how to proceed with the material.

I'm trying to make a page that lists all the materials and where if the technician consumes a material item press minus and plus when getting new material... with on each line a total of how many he has in his possession.

On a later basis I want to offer the possibility so that I or the tech can set a minimum that he should have / defined per user and item. provided that some regions consume more items than others.

the home screen will then show a list of the materials that are running out.

is this possible?

Sincerely, B
![userstock](https://user-images.githubusercontent.com/17248804/133195629-d6e2b696-5440-4987-b580-04b1f721baa2.png)

**Comment by mjashanks** [comment]
Hi @bdevroey .

I have almost completed your requirement. The only thing that I could not achieve right now is to have a separate `Stock Items` table.

The video below hopefully highlights the key points. Additionally, I have attached an export of my app ("stock management.txt"), which you can import to see what I've done.

[stock_management.txt](https://github.com/Budibase/budibase/files/7161281/stock_management.txt)

https://user-images.githubusercontent.com/3524181/133245711-c6c679dd-6989-4ea4-ab88-3baebe237533.mp4

**Reply by mjashanks** [reply]
if you need to have a separate stock items table, you can still do that, it just won't be linked as a relationship to the `User Stock` table - I encountered an issue in trying this. I'm not sure if this limits you or not - if you give me some extra details, I can try and come up with a solution.

**Reply by D4RKNOISE** [reply]
Wow, thanks for the fast detailed response... already a big thanks for this!

To answer you question to have more details...

The goal i want to reach is:
Predifined Stock items...
 - The the items already exists in a seperated table.
  - - I difined the Tables and collum's as 
      <Stock_Items>
         > Stock_ID (this way i can adjust the item name later on) 
         > Item_Name
         > Item_Categorie
         
      <Stock_Categories>
         > Stock_Categorie_ID (this way i can adjust the item name later on) 
         > Stock_Categorie_Name
         
      <User_Stock>
         > User_ID
         > User_Stock_Item_ID (<Stock_Items>)
         > User_Current_Stock (counter MIN/PLUS)
         > User_Item_Stock_warn_limit (will be added if counter gets less than a predefined value)
         
         So what i try to happen if i push + or - ... create a new row in <User_Stock> if not extists...
         Or Update Existing <User_Stock> row with
         Logged in User_ID, User_Stock_Item_ID and the changing value -> User_Current_Stock...
         
This is the detailed part about what i try to realise....

Kind Regards,
Bart

**Reply by mjashanks** [reply]
I get you - we **should** be able to do this, but cannot right now because of this bug: https://github.com/Budibase/budibase/issues/2284

I can create the same application as above, except use a `Stock Items` table (instead of an  `Options` column). However, it does not display the name of the `Stock Item`, because of the bug.

We should have this fixed soon.

**Reply by D4RKNOISE** [reply]
Ah, i understand what you ment after reading the #2284 bug...

Besides this, do you think it would be better for this project to store its on a mysql server instead of the internal Bb database? 
because the data can grow bigger fast?

kind regards,
Bart

**Reply by mjashanks** [reply]
The internal BB database shouldn't have a problem with growth - it's backed by CouchDB. 

But you will be able to get around this issue with MySQL.

**Reply by D4RKNOISE** [reply]
ah nice :) i'll have time :p

i also found this: https://github.com/Budibase/budibase/pull/2551 can this be the reason?
